File: model/src/main/java/pl/agh/harmonytools/model/MinorScale.java
package pl.agh.harmonytools.model;

import lombok.Getter;

@Getter
public class MinorScale extends Scale {

    private final int[] pitches = {0, 2, 3, 5, 7, 8, 10};
    private Key key;

    public MinorScale(Key key) {
        super(key);
    }

    public MinorScale(BaseNote basenote, int tonicPitch){
        super(basenote, tonicPitch);
    }
}

File: model/src/main/scala/pl/agh/harmonytools/model/MinorScale.scala
package pl.agh.harmonytools.model

case class MinorScale(private val minorKey: Key) extends Scale {
  override protected val key: Key = minorKey
  override protected val pitches: List[Int] = List(0, 2, 3, 5, 7, 8, 10)
}

object MinorScale {
  def apply(baseNote: BaseNote.BaseNoteType, tonicPitch: Int): MinorScale =
    MinorScale(Key(baseNote, tonicPitch))
}
